src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -263,25 +241,34 @@ public AnalysisResults(bool ignoreCase, bool allAsciiIfIgnoreCase, int hashIndex
             public bool RightJustifiedSubstring => HashIndex < 0;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ReadOnlySpan<char> Slicer(this string s, int index, int count) => s.AsSpan((index >= 0 ? index : s.Length + index), count);
```

Comment:
This ternary is the sole remaining conditional jump in the hot loop, but there's no good way to avoid that simultaneous avoiding `delegate` overhead so make it as simple as possible.

Reply:
I have tried running this as a jumpless method body, but it make little difference and is harder to grok (passing `fromRight = 0` for left, or `fromRight = 1` for right which requires the `SubstringComparer`s to carry the left/right multiplier with them (so more state...):
```csharp
public static ReadOnlySpan<char> Slicer(this string s, byte fromRight, int index, int count) => s.AsSpan((s.Length * fromRight) + index), count);
```

Also, if we COULD swap the `HashSet<string>`'s comparer out for left vs. right the we could just have that knowledge embedded with a trait and thus fully jumpless, but that would require allocating two `HashSet<string>`s which might be an allocation regression nobody wants :(

Reply:
I  did something like that in PR #89689 which is a huge win... since we can't swap the comparer, I ended up creating both a left and right HashSet with backing comparer that "hard codes" the left/right logic. HUGE WINS, see that PR.